### PR TITLE
REF use one osx script with miniforge

### LINKED ISSUE #1292: conda update being run after the ci install version is set on osx
The osx env is updated after the ci setup script package is installed so the 2.* is not respected.

### PR DESCRIPTION
**Pull request by beckermr**
This PR centralizes the OSX CI scripts to use a single build script with miniforge. 

closes #1292 

Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

File: conda_smithy/configure_feedstock.py [hunk 1]
```diff
@@ -919,11 +919,6 @@ def render_circle(jinja_env, forge_config, forge_dir, return_metadata=False):
             os.path.join(forge_dir, ".circleci", "checkout_merge_commit.sh"),
             os.path.join(forge_dir, ".circleci", "fast_finish_ci_pr_build.sh"),
         ],
-        "linux": [
-            os.path.join(forge_dir, ".scripts", "run_docker_build.sh"),
-            os.path.join(forge_dir, ".scripts", "build_steps.sh"),
-        ],
-        "osx": [os.path.join(forge_dir, ".circleci", "run_osx_build.sh")],
```
Comment (beckermr) [inline conda_smithy/configure_feedstock.py:926]: smithy now handles the deletion of these files at the start and then they are rendered on each provider in the `_render_template_exe_files`. thus I am removing them here and for other ci providers below.

File: conda_smithy/configure_feedstock.py [hunk 2]
```diff
@@ -994,6 +989,15 @@ def _render_template_exe_files(
...
                 if old_file_contents != new_file_contents:
+                    import difflib
+                    import sys
+                    print("diff:")
+                    sys.stdout.writelines(difflib.unified_diff(
+                        old_file_contents.splitlines(),
+                        new_file_contents.splitlines(),
+                        fromfile=target_fname,
+                        tofile=target_fname,
+                    ))
```
Comment (beckermr) [inline conda_smithy/configure_feedstock.py:1000]: helpful for debugging!

File: conda_smithy/configure_feedstock.py [hunk 3]
```diff
@@ -1581,13 +1576,13 @@ def clear_variants(forge_dir):
 
 
 def get_common_scripts(forge_dir):
-    for old_file in ["run_docker_build.sh", "build_steps.sh"]:
+    for old_file in ["run_docker_build.sh", "build_steps.sh", "run_osx_build.sh"]:
```
Comment (beckermr) [inline conda_smithy/configure_feedstock.py:1579]: makes sure that `_render_template_exe_files` checks the osx build script for different renderings on different platforms, not that that should ever happen.

File: conda_smithy/configure_feedstock.py [hunk 4]
```diff
@@ -1581,13 +1576,13 @@ def clear_variants(forge_dir):
...
         yield os.path.join(forge_dir, ".scripts", old_file)
 
 
 def clear_scripts(forge_dir):
-    for folder in [".azure-pipelines", ".circleci", ".drone", ".travis"]:
-        for old_file in ["run_docker_build.sh", "build_steps.sh"]:
+    for folder in [".azure-pipelines", ".circleci", ".drone", ".travis", ".scripts"]:
+        for old_file in ["run_docker_build.sh", "build_steps.sh", "run_osx_build.sh"]:
```
Comment (beckermr) [inline conda_smithy/configure_feedstock.py:1585]: cleaning it all out at once

File: conda_smithy/templates/azure-pipelines-osx.yml.tmpl [hunk 5]
```diff
@@ -21,62 +21,13 @@ jobs:
...
 {%- if upload_on_branch %}
       export UPLOAD_ON_BRANCH="{{ upload_on_branch }}"
 {%- endif %}
-      upload_package ./ "./{{ recipe_dir }}" ./.ci_support/${CONFIG}.yaml
-    displayName: Upload package
+      ./.scripts/run_osx_build.sh
+    displayName: Run OSX build
     env:
       BINSTAR_TOKEN: $(BINSTAR_TOKEN)
-    condition: and(succeeded(), not(eq(variables['UPLOAD_PACKAGES'], 'False')))
```
Comment (beckermr) [inline conda_smithy/templates/azure-pipelines-osx.yml.tmpl:82]: This condition is enforced in the run_osx_build script now

File: conda_smithy/templates/run_osx_build.sh.tmpl [hunk 6]
```diff
@@ -2,38 +2,58 @@
 
 set -x
 
-{{ fast_finish }}
```
Comment (beckermr) [inline conda_smithy/templates/run_osx_build.sh.tmpl:5]: We don't need this anymore here. Travsi and azure never used it or don't now. Circle has its own step before this one.

File: conda_smithy/configure_feedstock.py [hunk 7]
```diff
@@ -994,6 +989,18 @@ def _render_template_exe_files(
             with open(target_fname, "r") as fh:
                 old_file_contents = fh.read()
                 if old_file_contents != new_file_contents:
+                    import difflib
+                    import sys
+
+                    print("diff:")
+                    sys.stdout.writelines(
```
Comment (isuruf) [inline conda_smithy/configure_feedstock.py:996]: Can you make this a debug log?

Comment (beckermr) [reply conda_smithy/configure_feedstock.py]: done.

**Comment by beckermr** [comment]
OK @conda-forge/core @isuruf this one is ready for review!

File: conda_smithy/templates/run_osx_build.sh.tmpl [hunk 8]
```diff
@@ -2,38 +2,52 @@
...
-
-echo "Configuring conda." && echo -en 'travis_fold:start:configure_conda\\r'
-source ~/miniconda3/bin/activate root
-
-conda install -n root -c conda-forge --quiet --yes conda-forge-ci-setup=2 conda-build
+echo -e "\n\nInstalling a fresh version of Miniforge."
+if [[ ${CI} == "travis" ]]; then
+  echo -en 'travis_fold:start:install_miniforge\\r'
+fi
+MINIFORGE_URL="https://github.com/conda-forge/miniforge/releases/download/4.8.3-2"
```
Comment (isuruf) [inline conda_smithy/templates/run_osx_build.sh.tmpl:9]:
```suggestion
MINIFORGE_URL="https://github.com/conda-forge/miniforge/releases/latest/download"
```

**Comment by isuruf** [comment]
Minor comment. Otherwise, looks good. I assume this worked in a test feedstock.

**Comment by beckermr** [comment]
Yup. I tested this in combination with azure and travis in multiple ways and it appeared to build. I will commit your change and retest before merging.

**Comment by beckermr** [comment]
For reference, I made PRs 55 through 63 on https://github.com/conda-forge/cf-autotick-bot-test-package-feedstock/pulls with all combinations of osx and linux on circle, travis and azure. ppc64le and aarch64 were both on for travis and drone as well. All 9 of these PRs passed on the fork, passed on dev branches on master, and uploaded packages correctly.

**Comment by isuruf** [comment]
Thanks

**Pull request merged**